packages/arb-ts/src/lib/l2Bridge.ts

```diff
@@ -0,0 +1,206 @@
+
+    this.arbTokenBridge = ArbTokenBridgeFactory.connect(
+      arbTokenBridgeAddress,
+      l2Signer
+    )
+  }
+
+  public async withdrawETH(value: BigNumber, destinationAddress?: string) {
+    const address = destinationAddress || (await this.getWalletAddress())
+    return this.arbSys.withdrawEth(address, {
+      value,
+    })
+  }
+
+  public async withdrawERC20(
+    erc20l1Address: string,
+    amount: BigNumber,
+    destinationAddress?: string
+  ) {
+    const destination = destinationAddress || (await this.getWalletAddress())
+
+    const tokenData = await this.getAndUpdateTokenData(erc20l1Address)
+    const erc20TokenData = tokenData.ERC20
+
+    if (!erc20TokenData) {
+      throw new Error(
+        `Can't withdraw; ArbERC20 for ${erc20l1Address} doesn't exist`
+      )
+    }
+    return erc20TokenData.contract.withdraw(destination, amount)
+  }
+
+  public async withdrawERC777(
+    erc20l1Address: string,
+    amount: BigNumber,
+    destinationAddress?: string
+  ) {
+    const destination = destinationAddress || (await this.getWalletAddress())
+
+    const tokenData = await this.getAndUpdateTokenData(erc20l1Address)
+    const erc777TokenData = tokenData.ERC777
+
+    if (!erc777TokenData) {
+      throw new Error(
+        `Can't withdraw; ArbERC777 for ${erc20l1Address} doesn't exist`
+      )
+    }
+    return erc777TokenData.contract.withdraw(destination, amount)
+  }
+
+  public async updateAllL2Tokens() {
+    for (const l1Address in this.l2Tokens) {
+      await this.getAndUpdateTokenData(l1Address)
+    }
+    return this.l2Tokens
+  }
+
+  public async getAndUpdateTokenData(erc20L1Address: string) {
+    if (!this.l2Tokens[erc20L1Address]) {
+      this.l2Tokens[erc20L1Address] = {
+        ERC20: undefined,
+        ERC777: undefined,
+        CUSTOM: undefined,
+      }
+    }
+
+    const tokenData = this.l2Tokens[erc20L1Address] as L2TokenData // truthiness is ensured above
+    const walletAddress = await this.getWalletAddress()
+    const l2ERC20Address = await this.getERC20L2Address(erc20L1Address)
+    const l2ERC777Address = await this.getERC777L2Address(erc20L1Address)
+
+    if (!tokenData.ERC20) {
+      if ((await this.l2Provider.getCode(l2ERC20Address)).length > 2) {
+        const arbERC20TokenContract = await StandardArbERC20Factory.connect(
+          l2ERC20Address,
+          this.l2Signer
+        )
+        const balance = await arbERC20TokenContract.balanceOf(walletAddress)
+        tokenData.ERC20 = {
+          contract: arbERC20TokenContract,
+          balance,
+        }
+      } else {
+        console.info(
+          `Corresponding ArbERC20 for ${erc20L1Address} not yet deployed`
+        )
+      }
+    } else {
+      const arbERC20TokenContract = await StandardArbERC20Factory.connect(
+        l2ERC20Address,
+        this.l2Signer
+      )
+      const balance = await arbERC20TokenContract.balanceOf(walletAddress)
+      tokenData.ERC20.balance = balance
+    }
```

Comment:
This seems to be the same as the L1 side. It would be awesome if we could share code between them